Assets/MixedRealityToolkit.Providers/ObjectMeshObserver/SpatialObjectMeshObserver.cs

```diff
@@ -0,0 +1,336 @@
+        protected virtual GameObject ObservedObjectParent => observedObjectParent != null ? observedObjectParent : (observedObjectParent = SpatialAwarenessSystem?.CreateSpatialAwarenessObservationParent(Name));
+
+        /// <inheritdoc />
+        public override void ClearObservations()
+        {
+            if (IsRunning)
+            {
+                Debug.Log("Cannot clear observations while the observer is running. Suspending this observer.");
+                Suspend();
+            }
+
+            foreach (int id in Meshes.Keys)
+            {
+                RemoveMeshObject(id);
+            }
+
+            // Resend file observations when resumed.
+            sendObservations = true;
+        }
+
+        private int currentMeshId = 0;
+
+        /// <inheritdoc />
+        public override void Resume()
+        {
+            if (!sendObservations) { return; }
+
+            IsRunning = true;
+
+            // Get the collection of MeshFilters
+            if (spatialMeshObject != null)
+            {
+                MeshFilter[] meshFilters = spatialMeshObject.GetComponentsInChildren<MeshFilter>();
+                for (int i = 0; i < meshFilters.Length; i++)
+                {
+                    SpatialAwarenessMeshObject meshObject = SpatialAwarenessMeshObject.Create(
+                        meshFilters[i].sharedMesh,
+                        MeshPhysicsLayer,
+                        $"Spatial Object Mesh {currentMeshId}",
+                        currentMeshId,
+                        ObservedObjectParent);
+
+                    meshObject.GameObject.transform.localPosition = meshFilters[i].transform.position;
+                    meshObject.GameObject.transform.localRotation = meshFilters[i].transform.rotation;
+
+                    ApplyMeshMaterial(meshObject);
+
+                    meshes.Add(currentMeshId, meshObject);
+
+                    currentMeshId++;
+                }
+            }
+            sendObservations = false;
+        }
+
+        /// <inheritdoc />
+        public override void Suspend()
+        {
+            IsRunning = false;
+        }
+
+        private void RemoveMeshObject(int id)
+        {
+            SpatialAwarenessMeshObject meshObject = null;
+
+            if (meshes.TryGetValue(id, out meshObject))
+            {
+                // Remove the mesh object from the collection.
+                meshes.Remove(id);
+                if (meshObject != null)
+                {
+                    SpatialAwarenessMeshObject.Cleanup(meshObject);
+                }
+
+                // Send the mesh removed event
+                SpatialAwarenessSystem?.RaiseMeshRemoved(this, id);
+            }
+        }
+
+        #endregion IMixedRealitySpatialAwarenessObserver implementation
+
+        #region IMixedRealitySpatialAwarenessMeshObserver implementation
+
+        public SpatialAwarenessMeshDisplayOptions displayOption = SpatialAwarenessMeshDisplayOptions.Visible;
```

Comment:
The field here should not be public. There is already a public property, which is part of the interface. Field should be marked serializable if necessary and should be private.

Reply:
good catch... was supposed to be private